test/tests/functional/pbs_cray_smoketest.py

```diff
@@ -0,0 +1,189 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2017 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+from ptl.utils.pbs_crayutils import CrayUtils
+import os
+
+
+@tags('cray', 'smoke')
+class TestCraySmokeTest(TestFunctional):
+
+    """
+    Set of tests that qualifies as smoketest for Cray platform
+    """
+
+    def setUp(self):
+        if not self.du.get_platform().startswith('cray'):
+            self.skipTest("Test suite only meant to run on a Cray")
+        TestFunctional.setUp(self)
+
+    @tags('cray', 'smoke')
+    def test_cray_apstat(self):
+        """
```

Comment:
I have moved this test test_cray_apstat and the test test_cray_pbsnodes into one test called test_cray_no_queued_job.  I think that after setUp and no jobs queued this is still a valid regression test. It is a regression test because in the past we have seen a bug wherein no jobs were queued and yet I found that one node is reserved.

Reply:
When would that bug happen? Was the bug due to something PBS was doing?  Or was it more of a problem with the system (e.g. someone had a leftover ALPS reservation)?